src/Sulu/Bundle/AdminBundle/Resources/js/containers/List/adapters/IconAdapter.js

```diff
@@ -0,0 +1,64 @@
+// @flow
+import {observer} from 'mobx-react';
+import React from 'react';
+import FlatStructureStrategy from '../structureStrategies/FlatStructureStrategy';
+import DefaultLoadingStrategy from '../loadingStrategies/DefaultLoadingStrategy';
+import singleIconSelectStyle from '../../../components/SingleIconSelect/singleIconSelect.scss';
+import SingleIconComponent from '../../../components/SingleIconSelect/SingleIcon';
+import AbstractAdapter from './AbstractAdapter';
+
+@observer
+class IconAdapter extends AbstractAdapter {
+    static LoadingStrategy = DefaultLoadingStrategy;
+
+    static StructureStrategy = FlatStructureStrategy;
+
+    static icon = 'su-magic';
+
+    /**
+     * @param onItemSelectionChange
+     */
+    handleClick = (onItemSelectionChange: ?(rowId: string | number, selected?: boolean) => void) => {
+        return onItemSelectionChange;
+    };
```

Comment:
i think this should look something like this:
```suggestion
    handleClick = (iconId: string) => {
        this.props.onItemSelectionChange(iconId, this.selections.includes(iconId));
    };
```